shoov/modules/custom/shoov_repository/shoov_repository.module

```diff
@@ -37,7 +37,83 @@ function shoov_repository_node_presave($node) {
   $data = shoov_github_http_request("repos/$owner/$repo", $options);
 
   // @todo: Throw exception on error?
+  // or try again or set state invalid to this repo.
   if (!empty($data['id'])) {
     $wrapper->field_github_id->set($data['id']);
   }
 }
+
+/**
+ * Implements hook_node_insert().
+ *
+ * Check that vocabulary with name "Screenshots tags" exist for this repository
+ * otherwise create one and assign with repository.
+ *
+ * @param $node
+ *    Repository object.
+ */
+function shoov_repository_node_insert($node) {
+  if ($node->type != 'repository') {
+    // Not a repository node.
+    return;
+  }
+
+  if (shoov_repository_vocabulary_exist('Screenshots tags', $node)) {
+    // Vocabulary for screenshots tags already exist.
+    return;
+  }
+
+  shoov_repository_create_screenshot_tags_vocabulary($node);
+}
+
+/**
+ * Determine the vocabulary exist in repository.
+ *
+ * @param $vocabulary
+ *    The name of a vocabulary.
+ * @param $node
+ *    The repository object or the id of repository.
+ *
+ * @return int|false
+ *    The vid of vocabulary or false if vocabulary doesn't exist.
+ */
+function shoov_repository_vocabulary_exist($vocabulary, $repository) {
+  if ($repository_id = is_int($repository)) {
```

Comment:
It's not really check. This part of code allow you to use ID of Repository instead of repository object.